src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,84 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private double threshold;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.threshold = DEFAULT_LOAD_FACTOR * DEFAULT_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (size == threshold) {
+            resize();
+        }
+        Node<K, V> node = findNode(key);
+        if (node != null) {
+            node.value = value;
+        } else {
+            Node<K, V> newNode = new Node<>(key, value, table[index]);
+            table[index] = newNode;
+            size++;
+        }
```

Comment:
Index нужен только внутри else - значит его не нужно считать сразу.
```suggestion
        if (size == threshold) {
            resize();
        }
        Node<K, V> node = findNode(key);
        if (node != null) {
            node.value = value;
        } else {
            Node<K, V> newNode = new Node<>(key, value, table[getIndex(key)]);
            table[index] = newNode;
            size++;
        }
```